airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/jsonl_parser.py

```diff
@@ -22,7 +24,27 @@ async def infer_schema(
         stream_reader: AbstractFileBasedStreamReader,
         logger: logging.Logger,
     ) -> Dict[str, Any]:
-        ...
+        """
+        Infers the schema for the file by inferring the schema for each line, and merging
+        it with the previously-inferred schema.
+        """
+        inferred_schema = {}
+        read_bytes = 0
+
+        with stream_reader.open_file(file) as fp:
+            for line in fp:
+                if read_bytes < self.MAX_BYTES_PER_FILE_FOR_SCHEMA_INFERENCE:
+                    line_schema = self.infer_schema_for_record(json.loads(line))
+                    inferred_schema = merge_schemas(inferred_schema, line_schema)
+                    read_bytes += len(line)
+
+        if read_bytes > self.MAX_BYTES_PER_FILE_FOR_SCHEMA_INFERENCE:
+            logger.warning(
```

Comment:
nit: it's possible the whole file was read since we compute the number of bytes read after reading the line

Reply:
I've updated the parser to use pyarrow so have removed this check for now.